File: server-fastapi/app/schemas/kardex.py
from pydantic import BaseModel
from datetime import datetime


class KardexCreate(BaseModel):
    product_id: str
    movement_type: str  # sale, restock, adjustment, shrinkage
    quantity: int
    notes: str | None = None
    user_id: str | None = None
    reference_id: str | None = None


class KardexOut(BaseModel):
    id: str
    product_id: str
    movement_type: str
    quantity: int
    stock_before: int
    stock_after: int
    reference_id: str | None
    notes: str | None
    user_id: str | None
    created_at: datetime

    model_config = {"from_attributes": True}

File: mobile-pos/src/api/kardex.ts
import { api } from './client'

export type KardexMovementType = 'restock' | 'adjustment' | 'shrinkage'

export interface KardexCreate {
  product_id: string
  movement_type: KardexMovementType
  quantity: number
  notes?: string
}

export interface KardexOut {
  id: string
  product_id: string
  movement_type: string
  quantity: number
  stock_before: number
  stock_after: number
  notes?: string | null
  created_at: string
}

export async function adjustStock(payload: KardexCreate): Promise<KardexOut> {
  const { data } = await api.post<KardexOut>('/kardex/', payload)
  return data
}
